compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -810,8 +810,8 @@ private boolean isAssignable(BType source, BType target, Set<TypePair> unresolve
             return true;
         }
 
-        if (TypeTags.isXMLTypeTag(sourceTag) && (TypeTags.isXMLTypeTag(targetTag) || targetTag == TypeTags.STRING)) {
-            return isXMLTypeAssignable(source, target, unresolvedTypes);
+        if (TypeTags.isXMLTypeTag(sourceTag) && isXMLTypeAssignable(source, target, unresolvedTypes)) {
```

Comment:
If the target type is an XML type or string when the source type is XML, if `isXMLTypeAssignable` returns false, we don't have to do any of the subsequent checks, right? By moving the `isXMLTypeAssignable` check to the condition aren't we introducing some additional unnecessary checks?

Reply:
Will re-introduce the short circuit